### ISSUE TITLE
Bug report: JVM crashes if saving EXR image sequence with color buffer of `ImageFormat.FLOAT[16|32]` type

### ISSUE THREAD

**Issue by sidec** [openrndr/openrndr#278]
### Operating System

Windows, Linux

### OPENRNDR version

0.3.58

### Java version

11, 15

### Describe the Bug

Hi I encountered this bug when I wanted to save an image sequence in EXR format. In short, the JVM crashes after a few minutes. Below is a message from the terminal when running the program in Windows 11. Same result in Linux, however do not have access to the machine anymore, will try to reproduce later on archlinux. Full log from Windows can attach in case of not being able to reproduce the behaviour but in essence this is what I got:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x00007fff57ba8246, pid=11612, tid=24172
#
# JRE version: OpenJDK Runtime Environment Zulu15.28+51-CA (15.0.1+9) (build 15.0.1+9)
# Java VM: OpenJDK 64-Bit Server VM Zulu15.28+51-CA (15.0.1+9, mixed mode, sharing, tiered, compressed oops, g1 gc, windows-amd64)
# Problematic frame:
# C  [ntdll.dll+0x28246]
#
# No core dump will be written. Minidumps are not enabled by default on client versions of Windows

---------------  S U M M A R Y ------------

Command Line: -javaagent:C:\Users\***\AppData\Local\JetBrains\Toolbox\apps\IDEA-C\ch-0\213.5744.223\lib\idea_rt.jar=53054:C:\Users\***\AppData\Local\JetBrains\Toolbox\apps\IDEA-C\ch-0\213.5744.223\bin -Dfile.encoding=UTF-8 TemplateProgramKt

Host: Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, 8 cores, 31G,  Windows 10 , 64 bit Build 22000 (10.0.22000.318)
Time: Tue Dec 21 18:30:06 2021 Romance Standard Time elapsed time: 59.073581 seconds (0d 0h 0m 59s


---------------  S Y S T E M  ---------------

OS:
Windows 10 , 64 bit Build 22000 (10.0.22000.318)
OS uptime: 0 days 6:45 hours
HyperV virtualization detected

CPU: total 8 (initial active 8) (4 cores per cpu, 2 threads per core) family 6 model 142 stepping 12 microcode 0xec, cmov, cx8, fxsr, mmx, sse, sse2, sse3, ssse3, sse4.1, sse4.2, popcnt, vzeroupper, avx, avx2, aes, clmul, erms, 3dnowpref, lzcnt, ht, tsc, tscinvbit, bmi1, bmi2, adx, fma, clflush, clflushopt

Memory: 4k page, system-wide physical 32094M (7752M free)
TotalPageFile size 36958M (AvailPageFile size 6700M)
current process WorkingSet (physical memory assigned to process): 4229M, peak: 4239M
current process commit charge ("private bytes"): 4569M, peak: 4581M

vm_info: OpenJDK 64-Bit Server VM (15.0.1+9) for windows-amd64 JRE (15.0.1+9) (Zulu15.28+51-CA), built on Nov  7 2020 08:05:57 by "zulu_re" with MS VC++ 15.9 (VS2017)

```

### Steps to Reproduce

Using the latest `openrndr-template`

```kotlin

import org.openrndr.application
import org.openrndr.color.ColorRGBa
import org.openrndr.draw.*
import java.nio.file.Files
import java.nio.file.Paths
import kotlin.math.cos
import kotlin.math.sin

fun main() = application {
    configure {
        width = 768
        height = 576
    }

    program {
        val image = loadImage("data/images/pm5544.png")
        val font = loadFont("data/fonts/default.otf", 64.0)

        val rt = renderTarget(width, height) {
            colorBuffer(type = ColorType.FLOAT32)
        }

        val rd = Files.createDirectories(Paths.get("","tmp").toAbsolutePath())

        extend {

            drawer.isolatedWithTarget(rt) {
                drawStyle.colorMatrix = tint(ColorRGBa.WHITE.shade(0.2))
                image(image)

                fill = ColorRGBa.PINK
                circle(cos(seconds) * width / 2.0 + width / 2.0, sin(0.5 * seconds) * height / 2.0 + height / 2.0, 140.0)

                fontMap = font
                fill = ColorRGBa.WHITE
                text("OPENRNDR", width / 2.0, height / 2.0)
            }

            drawer.image(rt.colorBuffer(0))

            val imageFileName = "%05d.exr".format(frameCount)
            val imageFile = rd.resolve(imageFileName).toFile()
            rt.colorBuffer(0).saveToFile(imageFile, ImageFileFormat.EXR, false)
            println("$imageFileName saved")
        }
    }
}

```

**Comment by sidec** [comment]
It seems to be an issue with `TinyEXR` and crash during a garbage collection. I reproduced the behaviour on linux machine with the same jvm configuration. I got in addition `free() invalide pointer` printed to console. Also I found  this [issues/699](https://github.com/LWJGL/lwjgl3/issues/699) on lwjgl git repo that might not be directly related to the present problem but can provide some hints . From inspecting openrndr.core I can see that `org.openrndr.internal.gl3::saveToFile` uses `*Buffer::allocateDirect`  or `BufferUtils::create*Buffer` methods.  @edwinRNDR It be beneficial to replace it with `org.lwjgl.system.MemoryUtil` equivalents or this is as it is by design?

**Comment by edwinRNDR** [comment]
I think the problem may be similar to https://github.com/LWJGL/lwjgl3/issues/535. Can be that some of the ByteBuffers are GC'd before `SaveEXRImageToFile` is called.

**Comment by edwinRNDR** [comment]
Problem exists also in OR 0.4

**Comment by sidec** [comment]
That's tricky, not sure which of buffers might be a culprit. Is calling FreeEXRImage [here](https://github.com/openrndr/openrndr/blob/92a909c2291ff0b41d5e8558baa5345aa3166bbf/openrndr-jvm/openrndr-gl3/src/jvmMain/kotlin/org/openrndr/internal/gl3/ColorBufferGL3.kt#L957) necessary? Prevent it from memory leak or gc will handle it automatically? Locking/blocking on `SaveEXRImageToFile` might help IYO?

**Comment by edwinRNDR** [comment]
I tried calling rewind() on almost all of buffers at the end of the function (just to make sure the buffers are referenced and not tagged for gc) but the problem persists when saving a sequence. Maybe it can be pinpointed by calling System.gc() after buffer allocation (in hopes System.gc() actually forces garbage collection).

**Comment by edwinRNDR** [comment]
https://github.com/openrndr/openrndr/commit/e62f2a934253738ba90ea9a676df9a92e951324f fixes it for OPENRNDR 0.4

**Comment by sidec** [comment]
That's great that single `exrImage.images(null as PointerBuffer?)` was sufficient. The issue will stay opened till the OPENRNDR 0.4 release?
